From db07675be3e4ae72e95fa5fbdf49071dd6e56608 Mon Sep 17 00:00:00 2001
From: Dennis Schridde <devurandom@gmx.net>
Date: Fri, 4 Oct 2019 15:59:21 +0200
Subject: [PATCH] onig_sys/build.rs: Also search /usr/include for oniguruma.h,
 if we fail to find it via pkg-config

Under some conditions (pkg-config 0.3.15?) the pkg-config crate presents
us with an empty list of include directories.  This patch works around it.
---
 onig_sys/build.rs | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/onig_sys/build.rs b/onig_sys/build.rs
index 779b7cc..358bde7 100644
--- a/onig_sys/build.rs
+++ b/onig_sys/build.rs
@@ -217,8 +217,16 @@ pub fn main() {
                         return
                     }
                 }
+                let c_search_paths = [Path::new("/usr/include")];
+                for path in &c_search_paths {
+                    let header = path.join("oniguruma.h");
+                    if header.exists() {
+                        bindgen_headers(&header.display().to_string());
+                        return;
+                    }
+                }
                 if require_pkg_config {
-                    panic!("Unable to find oniguruma.h in include paths from pkg-config: {:?}", lib.include_paths);
+                    panic!("Unable to find oniguruma.h in /usr/include and include paths from pkg-config: {:?}", lib.include_paths);
                 }
             },
             Err(ref err) if require_pkg_config => {
-- 
2.23.0

